pinecone/data/index.py

```diff
@@ -168,7 +154,7 @@ def upsert(
 
         Returns: UpsertResponse, includes the number of vectors upserted.
         """
-        _check_type = kwargs.pop("_check_type", False)
```

Comment:
I suppose this is a breaking change also. Type checking seems like something we want to happen by default.